VideoLocker/src/main/java/org/edx/mobile/util/observer/AsyncCallableUtils.java

```diff
@@ -0,0 +1,37 @@
+package org.edx.mobile.util.observer;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+
+public enum AsyncCallableUtils {
+    ;
+
+    // Using AsyncTask.THREAD_POOL_EXECUTOR to automatically support Espresso idling resources
+    @VisibleForTesting
+    @NonNull
+    public static Executor EXECUTOR = AsyncTask.THREAD_POOL_EXECUTOR;
```

Comment:
How about changing the executor when running tests on Espresso, as you have
done for the Robolectric BaseTest class? Also, doing a quick search on
this, it looks like it may possible to implement a
custom ThreadPoolIdlingResource:
https://stefanodacchille.github.io/blog/2014/04/05/wait-for-it/

On Thu, Mar 17, 2016 at 11:05 PM, Brian Guertin <notifications@github.com>
wrote:

> In
> VideoLocker/src/main/java/org/edx/mobile/util/observer/AsyncCallableUtils.java
> <https://github.com/edx/edx-app-android/pull/613#discussion_r56550782>:
>
> > +package org.edx.mobile.util.observer;
> > +
> > +import android.os.AsyncTask;
> > +import android.support.annotation.NonNull;
> > +import android.support.annotation.VisibleForTesting;
> > +
> > +import java.util.concurrent.Callable;
> > +import java.util.concurrent.Executor;
> > +
> > +public enum AsyncCallableUtils {
> > +    ;
> > +
> > +    // Using AsyncTask.THREAD_POOL_EXECUTOR to automatically support Espresso idling resources
> > +    @VisibleForTesting
> > +    @NonNull
> > +    public static Executor EXECUTOR = AsyncTask.THREAD_POOL_EXECUTOR;
>
> Per my comment, I'm using this thread pool because it is automatically
> compatible with Espresso.
>
> Once we switch to async retrofit / okhttp calls (which I would like to do
> ASAP), we won't need this thread pool any longer.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/613/files/1b79f86bec593c27df395eb26a9eb9586f474ec4#r56550782>
>